core/src/test/scala/org/apache/spark/sql/delta/cdc/MergeCDCSuite.scala

```diff
@@ -187,6 +193,118 @@ class MergeCDCSuite extends MergeIntoSQLSuite with DeltaColumnMappingTestUtils {
     confs = (DeltaSQLConf.DELTA_SCHEMA_AUTO_MIGRATE.key, "true") :: Nil
   )
 
+
+  testMergeCdcUnlimitedClauses("unconditional delete only with duplicate matches")(
+    target = Seq(0, 1).toDF("value"),
+    source = Seq(1, 1).toDF("value"),
+    mergeCondition = "t.value = s.value",
+    clauses = MergeClause(isMatched = true, null, "DELETE") :: Nil,
+    expectedTableData = Seq(0).toDF(),
+    expectedCdcData = ((1, "delete", 1) :: Nil).toDF()
+  )
+
+  testMergeCdcUnlimitedClauses(
+    "unconditional delete only with duplicate matches without duplicates in the source")(
+    target = Seq(0).toDF("value"),
+    source = ((0, 0) :: (0, 1) :: Nil).toDF("col1", "col2"),
```

Comment:
nit: any need for second column here?

Reply:
also .. this is the case duplicates in the source...but the test names says different

Reply:
This is to contrast with the first test case. Here there aren't any duplicate rows in the source, just duplicate matches. So if for example you were to solve this problem by deduping the source before the join... this test case would fail.

I mean so would the ones with insert below... But I consider this a slightly different case.

Reply:
slight overkill.. but i will allow it :)